### PR TITLE
[FIX] update examples with new altas API

### LINKED ISSUE #3547: Update examples that use old atlas API to showcase new behavior
After merge #3353, examples that use the now deprecated API for 3 atlases need to be updated.

Effected examples:
- examples/00_tutorials/plot_3d_and_4d_niimg.py
- examples/01_plotting/plot_multiscale_parcellations.py
- examples/01_plotting/plot_prob_atlas.py

### PR DESCRIPTION
**Pull request by achamma723**
Closes #3547 
This PR updates ```examples/00_tutorials/plot_3d_and_4d_niimg.py ```, ```examples/01_plotting/plot_multiscale_parcellations.py``` and ```examples/01_plotting/plot_prob_atlas.py``` that use the now deprecated API for 3 atlases

### CODE REVIEW

File: examples/01_plotting/plot_multiscale_parcellations.py [hunk 1]
```diff
@@ -19,12 +19,17 @@
 # import datasets module and use `fetch_atlas_basc_multiscale_2015` function
 from nilearn import datasets
 
-parcellations = datasets.fetch_atlas_basc_multiscale_2015(version="sym")
+parcellations = [datasets.fetch_atlas_basc_multiscale_2015(version='sym',
```
Comment (bthirion) [inline examples/01_plotting/plot_multiscale_parcellations.py:22]:
```suggestion
parcellations = [
   datasets.fetch_atlas_basc_multiscale_2015(version='sym', resolution=64),
``` etc.

File: examples/01_plotting/plot_multiscale_parcellations.py [hunk 2]
```diff
@@ -19,12 +19,17 @@
...
+                 resolution=197),
+                 datasets.fetch_atlas_basc_multiscale_2015(version='sym',
+                 resolution=444)]
 
 # We show here networks of 64, 197, 444
-networks_64 = parcellations["scale064"]
-networks_197 = parcellations["scale197"]
-networks_444 = parcellations["scale444"]
+networks_64 = parcellations[0]['map']
+networks_197 = parcellations[0]['map']
```
Comment (bthirion) [inline examples/01_plotting/plot_multiscale_parcellations.py:31]: There seems to be a bug here

File: examples/01_plotting/plot_prob_atlas.py [hunk 3]
```diff
@@ -53,27 +64,26 @@
...
-    "Smith2009 20 RSNs": smith.rsn20,
-    "Smith2009 70 RSNs": smith.rsn70,
-    "Smith2009 20 Brainmap": smith.bm20,
-    "Smith2009 70 Brainmap": smith.bm70,
-    "ICBM tissues": (icbm["wm"], icbm["gm"], icbm["csf"]),
-    "Allen2011": allen.rsn28,
-    "Pauli2017 Subcortical Atlas": subcortex.maps,
-    f"DiFuMo dimension {dim} resolution {res}": difumo.maps,
-}
+atlas_types = {'Harvard_Oxford': harvard_oxford.maps,
```
Comment (bthirion) [inline examples/01_plotting/plot_prob_atlas.py:67]:
```suggestion
atlas_types = {
    'Harvard_Oxford': harvard_oxford.maps,
``` etc.

File: examples/01_plotting/plot_prob_atlas.py [hunk 4]
```diff
@@ -53,27 +64,26 @@
...
 
 for name, atlas in sorted(atlas_types.items()):
     plotting.plot_prob_atlas(atlas, title=name)
 
 # An optional colorbar can be set
-plotting.plot_prob_atlas(
-    smith.bm10, title="Smith2009 10 Brainmap (with" " colorbar)", colorbar=True
-)
-print("ready")
+plotting.plot_prob_atlas(smith_bm10, title='Smith2009 10 Brainmap (with'
```
Comment (bthirion) [inline examples/01_plotting/plot_prob_atlas.py:85]: call arguments on multiple lines.
You should probably update your linters.

**Review by bthirion — COMMENTED** [review]
A few fixes needed --- should be simple.

**Review by bthirion — APPROVED** [review]
LGTM, thx.

File: examples/01_plotting/plot_prob_atlas.py [hunk 5]
```diff
@@ -57,23 +74,23 @@
...
+    "Smith 2009 10 RSNs": smith_rsn10,
+    "Smith2009 20 RSNs": smith_rsn20,
+    "Smith2009 70 RSNs": smith_rsn70,
+    "Smith2009 20 Brainmap": smith_bm20,
+    "Smith2009 70 Brainmap": smith_bm70,
     "ICBM tissues": (icbm["wm"], icbm["gm"], icbm["csf"]),
     "Allen2011": allen.rsn28,
     "Pauli2017 Subcortical Atlas": subcortex.maps,
-    f"DiFuMo dimension {dim} resolution {res}": difumo.maps,
+    "DiFuMo dimension {0} resolution {1}".format(dim, res): difumo.maps,
```
Comment (ymzayek) [inline examples/01_plotting/plot_prob_atlas.py:85]: Why this change? It's better to use f-string: https://peps.python.org/pep-0498/

**Review by ymzayek — COMMENTED** [review]
LGTM otherwise! I think you can also add a whatsnew entry for this one

File: doc/changes/latest.rst
Comment (ymzayek) [inline doc/changes/latest.rst:1]: @achamma723 not sure why another entry was changed here. You don't need to black this file.

File: doc/changes/latest.rst [hunk 6]
```diff
@@ -48,10 +48,18 @@ Changes
...
+  :func:`~regions.img_to_signals_labels`,
+  :func:`~regions.signals_to_img_labels`, :func:`~regions.img_to_signals_maps`
+  :func:`~regions.signals_to_img_maps` (:gh:`3523` by `Rémi Gau`_ and `Christian
+  Gerloff`_).
 
 - Moved packaging from ``setup.py`` and setuptools build backend to ``pyproject.toml`` and hatchling backend. This change comes about as new standards are defined for Python packaging that are better met by the new configuration (:gh:`3635` by `Yasmin Mzayek`_).
 
+- Examples are updated with the new atlas API :ref:`plot_3d_and_4d_niimg.py`,
+  :ref:`plot_multiscale_parcellations.py` and :ref:`plot_prob_atlas.py`with
+  new altas API (:gh:`3557` by `Ahmad Chamma`_).
```
Comment (ymzayek) [inline doc/changes/latest.rst:61]:
```suggestion
- Examples are updated with the new atlas API :ref:`sphx_glr_auto_examples_00_tutorials_plot_3d_and_4d_niimg.py`, :ref:`sphx_glr_auto_examples_01_plotting_plot_multiscale_parcellations.py` and :ref:`sphx_glr_auto_examples_01_plotting_plot_prob_atlas.py` (:gh:`3557` by `Ahmad Chamma`_).
```

**Review by ymzayek — APPROVED** [review]
@achamma723 the references need to be fixed, otherwise it's good to go!

**Comment by ymzayek** [comment]
Merging
